File: frontend/lib/models/matchingSettings.ts
export type ByDateStrategySettings = {
  days: number;
};

export type ByRatingStrategySettings = {};

export type FinetuneStrategySettings = {
  fraction: number;
};

export type RandomStrategySettings = {};

export type MatchingSettings = {
  by_date_strategy: ByDateStrategySettings | null;
  by_rating_strategy: ByRatingStrategySettings | null;
  finetune_strategy: FinetuneStrategySettings | null;
  random_strategy: RandomStrategySettings | null;
};

File: backend/vidrank/lib/models/matching_settings.py
from typing import Optional

from pydantic import BaseModel


class ByDateStrategySettings(BaseModel):
    """By date strategy settings model."""

    days: int


class ByRatingStrategySettings(BaseModel):
    """By rating strategy settings model."""


class FinetuneStrategySettings(BaseModel):
    """Finetune strategy settings model."""

    fraction: float


class RandomStrategySettings(BaseModel):
    """Random strategy settings model."""


class MatchingSettings(BaseModel):
    """Matching settings model."""

    by_date_strategy: Optional[ByDateStrategySettings]
    by_rating_strategy: Optional[ByRatingStrategySettings]
    finetune_strategy: Optional[FinetuneStrategySettings]
    random_strategy: Optional[RandomStrategySettings]
